design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,421 @@
+                      attribute_type:
+                        description: The type of setting.
+                        enum:
+                        - Enumeration
+                        - String
+                        - Integer
+                        - Boolean
+                        - Password
+                        type: string
+                      lower_bound:
+                        description: The lowest value for an Integer type
+                        type: integer
+                      max_length:
+                        description: Maximum length for a String type
+                        type: integer
+                      min_length:
+                        description: Minimum length for a String type
+                        type: integer
+                      read_only:
+                        description: Whether or not this setting is read only
+                        type: boolean
+                      reset_required:
+                        description: Whether or not a reset is required
+                        type: boolean
+                      unique:
+                        description: Whether or not setting's value is unique
+                        type: boolean
+                      upper_bound:
+                        description: The highest value for an Integer type
+                        type: integer
+                    type: object
+```
+
+### Risks and Mitigations
+
+There is no risk in exposing the majority of the BIOS settings even if these
+values are ReadOnly, such as `NumCores` or `CoreSpeed`. However, some vendors
+use the `Password` AttributeType defined in
+[Registry](https://redfish.dmtf.org/schemas/v1/AttributeRegistry.v1_3_5.json)
+and include settings like `SysPassword`. Although the current values are not
+included, or set to "", for these types, the values should not be stored in
+the status section and they should not be allowed to be set via the spec
+section.
+
+### Work Items
+
+Ironic
+
+- Cache BIOS Registry
+- Add support for BIOS Registry information in the Ironic API
+
+Gophercloud
+
+- Add support for nodes/{node}/bios endpoint.  The microversion
+  should be 1.74 in order to use the ``?detail=True`` field to
+  get the bios registry information.
+
+Baremetal-operator
+
+- Add a new CRD for `HostFirmwareSettings`
+- Add a new CRD for `FirwmareSchema``
+- Get BIOS Settings via the Ironic API at the end of the Registration state and
+  store the name/value pairs in `settingsData`
+- Determine if a schema already exists for the BIOS registry data received
+  in the Ironic API. Create a new `FirmwareSchema` CR if one doesn't exist
+  and update the reference in `settingsData`
+- Copy name/value pairs from `settingsData` to
+ `vendorSettings` for settings that are not `read-only` or `unique`
+- Check for changes to `vendorSettings` and when detected, validate
+  values and add updated name/value pairs to Ironic clean-steps API
+
+## Alternatives
```

Comment:
Is there any suggestion for the [`Tests Plan`](https://github.com/metal3-io/metal3-docs/blob/master/design/_template.md#test-plan)?

Reply:
Added a section for `Test Plan`